Decred Wallet/Features/Receive/ReceiveViewController.swift

```diff
@@ -0,0 +1,177 @@
+//  ReceiveViewController.swift
+//  Decred Wallet
+//
+// Copyright (c) 2018-2020 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+import Foundation
+import UIKit
+import Dcrlibwallet
+
+class ReceiveViewController: UIViewController, UIDocumentInteractionControllerDelegate {
+    @IBOutlet weak var moreMenuButton: UIButton!
+    @IBOutlet weak var mainContentViewHolder: RoundedView!
+    @IBOutlet weak var selectedAccountView: UIView!
+    @IBOutlet weak var accountNameLabel: UILabel!
+    @IBOutlet weak var walletNameLabel: UILabel!
+    @IBOutlet weak var totalAccountBalanceLabel: UILabel!
+    @IBOutlet private var addressQRCodeImageView: UIImageView!
+    @IBOutlet weak var walletAddressLabel: UILabel!
+    @IBOutlet weak var syncInProgressLabel: UILabel!
+
+    var tapGesture = UITapGestureRecognizer()
+    var oldAddress = ""
+
+    var selectedWallet: DcrlibwalletWallet?
+    var selectedAccount: DcrlibwalletAccount?
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.setupUI()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        self.checkSyncStatus()
+    }
+
+    func setupUI() {
+        self.addressQRCodeImageView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.copyAddress)))
+        self.walletAddressLabel.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.copyAddress)))
+        self.selectedAccountView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.showAccountSelectorDialog)))
+    }
+
+    @objc func copyAddress() {
+        DispatchQueue.main.async {
+            UIPasteboard.general.string = self.walletAddressLabel.text!
+            Utils.showBanner(parentVC: self, type: .success, text: LocalizedStrings.walletAddrCopied)
+        }
+    }
+
+    @objc func showAccountSelectorDialog(_ sender: Any) {
+        AccountSelectorDialog.show(sender: self,
+                                 title: LocalizedStrings.receivingAccount,
+                                 selectedWallet: selectedWallet,
+                                 selectedAccount: self.selectedAccount,
+                                 callback: self.updateSelectedAccount)
+    }
+
+    private func checkSyncStatus() {
```

Comment:
- This function does not check sync status, needs a rename, probably `displayAddressForFirstWalletAccount`.
- Please re-arrange the functions in this file to match how they are called. This `checkSyncStatus` function, for example should be positioned immediately after the `setupUI` function.